mne/io/egi/egi.py

```diff
@@ -65,79 +65,62 @@ def my_fread(*x, **y):
         info['event_codes'] = np.array(info['event_codes'])
     else:
         raise NotImplementedError('Only continous files are supported')
-
     info.update(dict(precision=precision, unsegmented=unsegmented))
-
+    info['dtype'], info['bytesize'] = \
+        {2: ('>i2', 2), 4: ('>f4', 4),
+         6: ('>f8', 8)}[info['precision']]
     return info
 
 
 def _read_events(fid, info):
     """Read events"""
-    unpack = [info[k] for k in ['n_events', 'n_segments', 'n_channels']]
-    n_events, n_segments, n_channels = unpack
-    n_samples = 1 if info['unsegmented'] else info['n_samples']
-    events = np.zeros([n_events, n_segments * info['n_samples']])
-    dtype, bytesize = {2: ('>i2', 2), 4: ('>f4', 4),
-                       6: ('>f8', 8)}[info['precision']]
-
-    info.update({'dtype': dtype, 'bytesize': bytesize})
-    beg_dat = fid.tell()
-
-    for ii in range(info['n_events']):
-        fid.seek(beg_dat + (int(n_channels) + ii) * bytesize, 0)
-        events[ii] = np.fromfile(fid, dtype, n_samples)
-        fid.seek(int((n_channels + n_events) * bytesize), 1)
-    return events
-
-
-def _read_data(fid, info):
-    """Aux function"""
-    if not info['unsegmented']:
-        raise NotImplementedError('Only continous files are supported')
-
+    events = np.zeros([info['n_events'],
+                       info['n_segments'] * info['n_samples']])
     fid.seek(36 + info['n_events'] * 4, 0)  # skip header
-    readsize = (info['n_channels'] + info['n_events']) * info['n_samples']
-    final_shape = (info['n_samples'], info['n_channels'] + info['n_events'])
-    data = np.fromfile(fid, info['dtype'], readsize).reshape(final_shape).T
-    return data
+    for si in range(info['n_samples']):
+        # skip data channels
+        fid.seek(info['n_channels'] * info['bytesize'], 1)
+        # read event channels
+        events[:, si] = np.fromfile(fid, info['dtype'], info['n_events'])
```

Comment:
... thus why I modified the `_read_events` function to loop through sample-by-sample, skipping over the channel data along the way. Should hopefully be fast enough since it shouldn't need to actually read that much data.